sdk/openai/Azure.AI.OpenAI/src/Custom/AzureOpenAIClient.cs

```diff
@@ -339,10 +338,25 @@ private static PipelinePolicy CreateAddUserAgentHeaderPolicy(AzureOpenAIClientOp
         });
     }
 
+    private static PipelinePolicy CreateAddClientRequestIdHeaderPolicy()
+    {
+        return new GenericActionPipelinePolicy(message =>
+        {
+            if (message?.Request?.Headers is not null)
+            {
+                string requestId = message.Request.Headers.TryGetValue(s_clientRequestIdHeaderKey, out string existingHeader) == true
+                    ? existingHeader
```

Comment:
curious - why do we need to set a x-ms-client-request-id via this custom policy rather than relying on the default value set on the [Request](https://github.com/Azure/azure-sdk-for-net/blob/09a650ffd29ed3d531224edb4d47cd51e8aeb9b5/sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.Request.cs#L36)

Reply:
Thanks, @christothes, and sorry for the delay!

In short: `Azure.Core` would (and did) handle it automatically, but `System.ClientModel` doesn't. They look quite similar otherwise, but for some of these Azure- or Microsoft-specific things we need to bring it back in manually.